tool/site/server_hapi/src/routes/dispense/withdraw.ts

```diff
@@ -0,0 +1,29 @@
+import Hapi from "@hapi/hapi"
+import {getSessionValue, removeFromSessionValue} from "../../services/session"
+import {Task} from "fhir/r4"
+import {getEpsClient} from "../../services/communication/eps-client"
+
+export default [
+  {
+    method: "POST",
+    path: "/dispense/withdraw",
+    handler: async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
+      const releaseRequest = request.payload as Task
+      const accessToken = getSessionValue("access_token", request)
+      const epsClient = getEpsClient(accessToken)
+      const withdrawResponse = await epsClient.makeWithdrawRequest(releaseRequest)
+      const withdrawRequestHl7 = await epsClient.makeConvertRequest(releaseRequest)
+      const success = withdrawResponse.statusCode === 200
+      if (success) {
+        removeFromSessionValue("released_prescription_ids", releaseRequest.groupIdentifier?.value, request)
```

Comment:
doing a withdraw should remove a dispense not a release right? is this name misleading?

Reply:
it's removing from released so that it displays in dispensed/partial dispense my-prescriptions table

Reply:
should already have been removed from released prescriptions by the first dispense request though, so probably don't need this here

Reply:
yeah I think that's a copy paste mistake

Reply:
if every dispense is withdrawn then we would remove it from dispense id array

Reply:
changed to removed from `"dispensed_prescription_ids"` instead